pkg/util/mount/mount_linux.go

```diff
@@ -944,6 +951,57 @@ func doCleanSubPath(mounter Interface, fullContainerDirPath, subPathIndex string
 	return nil
 }
 
+// cleanCorruptedMnt tears down the corrupted subpath bind mount
+func cleanCorruptedMnt(mounter Interface, fullSubPath string) error {
+	// process /var/lib/kubelet/pods/<uid>/volume-subpaths/<volume>/<container name>/<fullSubPath>
+	glog.V(4).Infof("Cleaning up subpath mounts for subpath %v", fullSubPath)
+	// Unmount it
+	if err := mounter.Unmount(fullSubPath); err != nil {
+		return fmt.Errorf("error unmounting %s: %s", fullSubPath, err)
+	}
+	glog.V(5).Infof("Unmounted %s", fullSubPath)
+	// Remove it *non*-recursively, just in case there were some hiccups.
+	if err := os.Remove(fullSubPath); err != nil {
+		return fmt.Errorf("error deleting %s: %s", fullSubPath, err)
+	}
+	glog.V(5).Infof("Removed %s", fullSubPath)
+	return nil
+}
+
+// pathExists returns true if the specified path exists.
+func pathExists(path string) bool {
```

Comment:
there is already `volumeutil.PathExists` func, could you use that?

Reply:
I have tried, but the compiler complaints `import cycle not allowed`

Reply:
I have tried, but the compiler complaints `import cycle not allowed`